lua/cmake-tools/quickfix.lua

```diff
@@ -99,44 +59,36 @@ function quickfix.run(cmd, env, args, opts)
     command = cmd,
     args = job_args,
     cwd = vim.loop.cwd(),
-    on_stdout = vim.schedule_wrap(append_to_quickfix),
+    on_stdout = vim.schedule_wrap(function(err, data)
+      append_to_quickfix(err, data)
+      on_output(data, err)
+    end),
     on_stderr = vim.schedule_wrap(function(err, data)
-      quickfix.notification.level = "warn"
       append_to_quickfix(err, data)
+      on_output(data, err)
     end),
     on_exit = vim.schedule_wrap(function(_, code, signal)
-      local msg = "Exited with code " .. (signal == 0 and code or 128 + signal)
-      local level = "error"
-      local icon = ""
+      code = signal == 0 and code or 128 + signal
+      local msg = "Exited with code " .. code
 
       append_to_quickfix(msg)
-
-      if code == 0 and signal == 0 then
-        level = quickfix.notification.level -- either info or warn
-        icon = ""
-        if opts.on_success then
-          opts.on_success()
-        end
-      elseif opts.cmake_quickfix_opts.show == "only_on_error" then
-        quickfix.show(opts.cmake_quickfix_opts)
+      if code == 0 and opts.show == "only_on_error" then
+        quickfix.show(opts)
         quickfix.scroll_to_bottom()
       end
```

Comment:
Compare to previous, I think this logic has problem. If opts.show equals only_an_error, we should only show quickfix window when code not equals zero